jetty-util/src/main/java/org/eclipse/jetty/util/Index.java

```diff
@@ -193,42 +226,55 @@
                 return this;
             }
 
+            public Builder<V> alphabet(String alphabetString)
+            {
+                alphabet = new HashSet<>();
+                alphabetString.chars().forEach(c -> alphabet.add((char)c));
+                return this;
+            }
+
+            public Builder<V> alphabet(Set<Character> alphabet)
+            {
+                this.alphabet = alphabet;
+                return this;
+            }
+
             /**
              * Build a {@link Mutable} instance.
              * @return a {@link Mutable} instance.
              */
             public Mutable<V> build()
             {
-                if (contents != null && maxCapacity == 0)
```

Comment:
Why did you get rid of this check? IMHO building an index both pre-loaded with some contents and a max capacity of zero looks like a programming mistake we want to catch.

Reply:
Then we need another check as returning an empty Trie wont catch programming errors